src/main/java/frc/robot/subsystems/swerve/module/maple/MapleModule.java

```diff
@@ -0,0 +1,123 @@
+package frc.robot.subsystems.swerve.module.maple;
+
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.controller.SimpleMotorFeedforward;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.wpilibj2.command.sysid.SysIdRoutine;
+import frc.robot.subsystems.swerve.module.Module;
+import frc.robot.subsystems.swerve.module.ModuleConstants;
+import frc.utils.calibration.sysid.SysIdCalibrator;
+import org.ironmaple.simulation.drivesims.SwerveModuleSimulation;
+import org.littletonrobotics.junction.AutoLog;
+import org.littletonrobotics.junction.Logger;
+
+public class MapleModule extends Module {
+
+	@AutoLog
+	public static class ModuleIOInputs {
+
+		public double drivePositionRad = 0.0;
+		public double driveVelocityRadPerSec = 0.0;
+		public double driveAppliedVolts = 0.0;
+		public double[] driveCurrentAmps = {0};
+
+		public Rotation2d turnAbsolutePosition = new Rotation2d();
+		public Rotation2d turnPosition = new Rotation2d();
+		public double turnVelocityRadPerSec = 0.0;
+		public double turnAppliedVolts = 0.0;
+		public double[] turnCurrentAmps = {0};
+
+	}
+
+	private final SwerveModuleSimulation moduleSimulation;
+	private final ModuleIOInputsAutoLogged moduleIOInputs;
+	private final PIDController steerPIDControllerRadians;
+	private final PIDController drivePIDControllerRadiansPerSecond;
+	private final SimpleMotorFeedforward driveFeedForwardRadiansPerSecond;
+
+	public MapleModule(ModuleConstants constants, MapleModuleConstants mapleModuleConstants, SwerveModuleSimulation moduleSimulation) {
+		super(constants);
+		this.moduleSimulation = moduleSimulation;
+		this.moduleIOInputs = new ModuleIOInputsAutoLogged();
+		this.steerPIDControllerRadians = mapleModuleConstants.steerPIDControllerRadians();
+		this.drivePIDControllerRadiansPerSecond = mapleModuleConstants.drivePIDControllerRadiansPerSecond();
+		this.driveFeedForwardRadiansPerSecond = mapleModuleConstants.driveFeedForwardRadiansPerSecond();
+	}
+
+	@Override
+	public SysIdCalibrator.SysIdConfigInfo getSteerSysIdConfigInfo() {
+		return new SysIdCalibrator.SysIdConfigInfo(new SysIdRoutine.Config(), false);
+	}
+
+	@Override
+	public SysIdCalibrator.SysIdConfigInfo getDriveSysIdConfigInfo() {
+		return new SysIdCalibrator.SysIdConfigInfo(new SysIdRoutine.Config(), false);
+	}
+
+	@Override
+	public void updateInputs() {
+		moduleIOInputs.drivePositionRad = moduleSimulation.getDriveWheelFinalPositionRad();
```

Comment:
I looked at their source code. Since the simulation uses 250Hz odometery for 50Hz roborio, I'd except to be able to get more than one value. `getCachedDriveEncoderUnGearedPositionsRad` should return a cache of the previous positions. It looks like the gear ratio isn't not applied on it since it caches the raw data from the encoder.